Support `--json` output for `status` command

AliSoftware opened the pull request:
## What

Allows to easily use the output in scripts if needed (e.g. pipe that status output through `jq`, etc)

## How

 - Introduce dedicated `struct RepositoryStatus` object to represent the status reported by `git-conceal status`
 - Introduce dedicated `struct FileStatusList` and `struct FileStatus` objects to represent the status of files list reported by `git-conceal status file1 file2 …`
 - Make those structs implement both the `Serialize` trait from the `serde` crate—to handle JSON serialization (using `#[derive(Serialize)]` for implicit implementation of it)—and the `fmt::Display` trait—to handle plain text rendering.
 - Which allows the implementation of `cmd_status` to build those objects representations regardless of the output format requested, and only at the end decide to `println!` those results either as JSON (`serde_json::to_string_pretty(&status)`) or as plain text (`print!("{}", status)`). (Making the code nicer than if I did giant `if / else` branches for all the cases)

## Testing Instructions

 - Run `cargo build --release` from this PR's head branch
 - Checkout the head branch of one of the PRs of a repo in which I've already prepared the migration to `git-conceal`, like of https://github.com/woocommerce/woocommerce-android/pull/14979 or https://github.com/Automattic/pocket-casts-android/pull/4841, so you have a repo on which to test that command on
 - Then, from the working copy of such a test repo, run some commands to test the feature, like:
```bash
$ export PATH="<path-to-git-conceal-repo>/target/release:$PATH"
$ git-conceal status --json | jq
$ git conceal status --json | jq -r '.encrypted_files[]'
$ git conceal status --json secret.properties | jq
$ git conceal status --json secret.properties foo bar baz | jq '.files | map({key:.file, value:.encrypted}) | from_entries'
```

AliSoftware committed:
Support `--json` output for `status` command

iangmaia commented on src/status.rs:
```diff
@@ -0,0 +1,79 @@
+#[derive(Serialize)]
+pub struct RepositoryStatus {
+    pub repository: String,
+    pub status: String,
```
Maybe this can be an enum? Something like:

```rust
#[derive(Serialize)]
#[serde(rename_all = "lowercase")] // see https://serde.rs/container-attrs.html#rename_all
pub enum LockStatus {
    Locked,
    Unlocked,
}
```

AliSoftware replied:
Good idea, I like it 👍

AliSoftware replied:
Implemented in 1427abf4f96e83e656883b1eb224725ebaf9bcd7

AliSoftware committed:
Using an `enum LockStatus` in `RepositoryStatus`

h/t @iangmaia in https://github.com/Automattic/git-conceal/pull/11#discussion_r2611457728

iangmaia reviewed (approved):
Code looks good 👍 and I ran the test instructions in [PCAndroid](https://github.com/Automattic/pocket-casts-android/pull/4841) and it's working as expected :shipit:
